BadHabbit/BadHabbit/Kit/NetworkKit/RequestExecutor.swift

```diff
@@ -30,13 +30,27 @@ class RequestExecutor: RequestExecutorProtocol {
     func perform<Parser: ResponseParserProtocol>(
         request: Request,
         parser: Parser,
-        completion: (Result<Parser.Response, Error>) -> Void
+        completion: @escaping (Result<Parser.Response, Error>) -> Void
     ) {
-        do {
-            let response = try parser.parse(Data())
-            completion(.success(response))
-        } catch {
-            completion(.failure(error))
+        if let url = URL(string: request.path) {
```

Comment:
In case if URL isn't correct we never call completion - that is bad behaviour. Completion must be called in this function

Reply:
Resolved